extensions/cloud/portability-cloud-microsoft/src/main/java/org/dataportabilityproject/cloud/microsoft/cosmos/AzureTableStore.java

```diff
@@ -0,0 +1,285 @@
+package org.dataportabilityproject.cloud.microsoft.cosmos;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.base.Preconditions;
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.azure.storage.table.CloudTable;
+import com.microsoft.azure.storage.table.CloudTableClient;
+import com.microsoft.azure.storage.table.TableOperation;
+import com.microsoft.azure.storage.table.TableQuery;
+import com.microsoft.azure.storage.table.TableResult;
+import org.dataportabilityproject.spi.cloud.storage.JobStore;
+import org.dataportabilityproject.spi.cloud.types.JobAuthorization;
+import org.dataportabilityproject.spi.cloud.types.PortabilityJob;
+import org.dataportabilityproject.types.transfer.models.DataModel;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.util.Iterator;
+import java.util.UUID;
+
+import static com.microsoft.azure.storage.table.TableQuery.generateFilterCondition;
+
+/** Uses the Azure Cosmos DB Table Storage API to persist job data. */
+public class AzureTableStore implements JobStore {
```

Comment:
this replaces CosmosStore, right? Anyway to make github realize it's a rename?

Reply:
Git doesn't explicitly track renames but instead computes it by comparing the SHA hashes of the file contents so that would be difficult. I also think this is not a rename since one class that provided a piece of functionality is being removed and another class added.

Reply:
interesting, did not know that